server/routes/admin/webhooks.py

```
@@ -0,0 +1,133 @@
+        return _json_response({"error": "Unknown webhook"}, 404)
+
+    # Require secret on all webhooks -- reject unsigned requests
+    if not hook.get("secret"):
+        return _json_response({"error": "Webhook has no secret configured"}, 403)
+
+    if not webhook_service.verify_incoming(payload, signature, hook["secret"]):
+        return _json_response({"error": "Invalid signature"}, 403)
+
+    try:
+        body = json.loads(payload)
+    except Exception:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)